### PR TITLE
Themes: Update Documentation

### PR DESCRIPTION
**Pull request by simeonoff**
Closes # 

### Checklist:
 
 - [ ] check topic's TOC/menu and paragraph headings
 - [ ] link to other topics using `../relative/path.md`
 - [ ] at the References section at the end of the topic add links to topics, samples, etc
 - [ ] reference API documentation instead of adding a section with API

<br />

 - [ ] use valid component names - [Data] Grid, `IgxSelectComponent`, `<igx-combo>`
 - [ ] use spell checker tool (VS Code, Grammarly, Microsoft Editor)
 - [ ] add inline `code blocks` for the names of classes / tags / properties
 - [ ] add language descriptor for the ```code blocks```
 - [ ] check broken links (use browser add-on)
 - [ ] check if sample is working and fully visible in the topic
 - [ ] check if sample is working and fully visible in the StackBlitz
 - [ ] check if code blocks match the code in StackBlitz demo

<br />

 - [ ] follow SEO guidelines to fill topic's metadata ([SEO guidelines](https://infragisticsinc297.sharepoint.com/:w:/g/Groups/marketing/EWz9InT4FDlErHCumxsKGY4Bd8H03yhRWxDFk47luRz-_Q?e=S5wWcx))

<br />

 - [ ] do not resolve requested changes (leave that to the reviewer)
 - [ ] add `pending-localization` label when the review of the PR is done
 - [ ] add a member from the localization team to translate it

### CODE REVIEW

File: en/components/themes/index.md [hunk 1]
```diff
@@ -1,61 +1,189 @@
...
+<!-- _description: The Ignite UI for Angular Theming component was developed through SASS with a low-difficulty API that offers restyling of one component, multiple components, or the entire suite. -->
+<!-- _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI widgets, Angular, Native Angular Components Suite, Native Angular Controls, Native Angular Components Library, Native Angular Components, Angular Theming Component, Angular Theming -->
+<!-- --- -->
+
+# Theming
+Ignite UI for Angular allows you to modify the styles of all component themes using CSS variables. If you really wanted to dig deep, we provide a powerful Sass theming engine that allows you to create global component themes tailored to your specific design language that work in both modern and old browsers.
+
+>[!NOTE] 
+> This document describes the theming system in Ignite UI for Angular from version 12 forward. If you are looking for the old documentation, visit [link](https://link-to-old-docs). Starting with version 12, we have extended support for CSS variables and they are now the recommended way to modify the theme.
```
Comment (SisIvanova) [inline en/components/themes/index.md:11]: The link has to be fixed.

File: en/components/themes/index.md [hunk 2]
```diff
@@ -1,61 +1,189 @@
...
+
+```css
+/* styles.css */
+:root {
+  --igx-primary-500: #7eaf6e;
+  --igx-secondary-500: #301764;
+}
+```
+
+Let's break down the names of these color variables. The `igx` prefix is there as a unique identifier to indicate that this variable is part of an Ignite UI for Angular theme, `primary` is the name of the `color` palette, and `500` is the color `variant`. We will take a deeper look at palettes in the [Palettes](https://something.com) section of the documentation. For now all you need to know is that we have several colors (primary, secondary, success, info, etc.) that include different shades or _variants_ that are all generated from the `500` color variant, i.e. the `500` variant is considered to be the main color variant. Some color variants are lighter while others are darker.
```
Comment (SisIvanova) [inline en/components/themes/index.md:83]: The link has to be fixed.

File: en/components/themes/index.md [hunk 3]
```diff
@@ -1,61 +1,189 @@
...
+
+Following this approach, you can completely overhaul the entire palette.
+
+>[!WARNING]
+> Some components do not use colors from the palettes. In those cases you will have to target the component CSS variables directly to modify their colors.
+> To find out which palette colors are used by what component, take a look at the [components documentation](https://link-to-docs).
```
Comment (SisIvanova) [inline en/components/themes/index.md:89]: The link has to be fixed.

File: en/components/themes/index.md [hunk 4]
```diff
@@ -1,61 +1,189 @@
...
+                    0 2px 1px -1px rgba(0, 0, 0, 0.08);
+  /* ... */
+  --igx-elevation-24: 0 11px 15px -7px rgba(0, 0, 0, 0.26),
+                  0 24px 38px 3px rgba(0, 0, 0, 0.12),
+                  0 9px 46px 8px rgba(0, 0, 0, 0.08);
+}
+```
 
-# Themes
+These are essentially stacked CSS [`box-shadow`](https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow) declarations. You can replace them with any other valid `box-shadow` value. The higher the elevation level number is, the bigger the shadow is. Again, different components use different elevation levels, to find out which elevation level a component uses take look at the [components documentation](link to Sass documentation). We will take a deeper look at elevations in the [Elevations](link to elevations section) of the documentation.
```
Comment (SisIvanova) [inline en/components/themes/index.md:107]: Components documentation and Elevations links are wrong

File: en/components/themes/sass/animations.md [hunk 5]
```diff
@@ -5,19 +5,92 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+//include the animation mixin with parameters
+.my-class {
+    @include animation('fade-in-top' 3s $ease-out-quad infinite);
+}
+```
+<div class="divider--half"></div>
+
+### Timing Functions
+
+We include a list of pre-baked timing functions to use with our keyframes mixins. Read the [documentation]({environment:sassApiUrl}/index.html) the full list of timing functions visit their documentation.
```
Comment (SisIvanova) [inline en/components/themes/sass/animations.md:68]:
```suggestion
We include a list of pre-baked timing functions to use with our keyframes mixins. Read the [documentation]({environment:sassApiUrl}/index.html) to find the full list of timing functions.
```

File: en/components/themes/sass/palettes.md [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+);
+
+@include igx-palette-vars($my-palette);
+```
+
+## API Reference
+* [Palettes]({environment:sassApiUrl}/index.html#function-igx-palette)
+* [Getting Palette Colors]({environment:sassApiUrl}/index.html#function-igx-color)
+* [Getting Contrast Colors]({environment:sassApiUrl}/index.html#function-igx-contrast-color)
+* [Generating Color Classes]({environment:sassApiUrl}/index.html#mixin-igx-color-classes)
```
Comment (SisIvanova) [inline en/components/themes/sass/palettes.md:300]: The last link is wrong

**Review by SisIvanova — COMMENTED** [review]
In the sass/component-themes.md file the Global Themes links at line 31 and 211 should be (./global-themes.md)

File: en/components/themes/sass/palettes.md (hunk 6 above)
Comment (simeonoff) [reply en/components/themes/sass/palettes.md:300]: This is related to the `@access` annotation in the source documentation in Ignite UI for Angular. It is set to private.

**Comment by randriova** [comment]
https://github.com/IgniteUI/igniteui-docfx/pull/2697

**Pull request merged**